src/pointer.rs

```diff
@@ -288,6 +248,36 @@ impl Pointer {
             )
             .into()
     }
+    /// Splits the `Pointer` at the given index if the character at the index is
```

Comment:
I wonder if it wouldn't be more useful to index by slash ordinal (rather than character index, which caller may have difficulty choosing?)

Reply:
Hmm, yea, that may make sense. 

I added it because I removed owned copies of pointers & tokens from the errors and went with offsets instead. I was debating whether a range of the token instead but just kept it simple.

Hmm.

Reply:
Either would work post-parsing but we'd still need `offset` for parsing-related errors. Speaking of which, I think I need to add the offset (or whatever we end up using) to `ParseError::InvalidEncoding` as I believe `offset` inside `InvalidEncodingError` is for the `Token`. Or perhaps I just don't re-use `InvalidEncodingError` there.

The only place where it will be a potential point of confusion is parsing if `ParseError` used offsets while `AssignError`, `ResolveError`, and `DeleteError` all used the token index. However, that may still be less confusing overall.

I honestly don't know what the better approach is here.

Reply:
Hmm I think there are a couple different cases here. When parsing it makes sense to use `offset` because at that point we still have a string, not a `Pointer`, and the offset is the correct way to point at where the error was originated. The same goes for `Token`, because there it's just another case of parsing.

Once parsing succeeded, though, errors refer to proper `Pointers`, which despite being internally represented as strings are a more structured type. It makes sense for APIs that manipulate pointers to use a different way to refer to positions, especially if they operate on a `Token` basis. I don't think there's any confusion.

Another approach would be to wrap the offset in a newtype, so we can hold the invariant that it points to a valid `/` boundary. That reduces the likelihood of misuse. If this only appears in the context of `Token` manipulation, a token index would be good too, but potentially a direct offset is more efficient in practice, since we can use to skip straight into the position in the internal string.

Yet another option, if the offset is intended so we can find a token in the `Pointer`, is to use a slice instead.

Reply:
> Once parsing succeeded, though, errors refer to proper Pointers, which despite being internally represented as strings are a more structured type. It makes sense for APIs that manipulate pointers to use a different way to refer to positions, especially if they operate on a Token basis. I don't think there's any confusion.

That's fair, totally on board with that line of thought.

> Another approach would be to wrap the offset in a newtype, so we can hold the invariant that it points to a valid / boundary. 

I like this approach but we'd need a set of methods to facilitate it so there is utility beyond error location -> token. It'd need to become a construct outside of that flow or it'll be an odd appendage.

I think whatever it is we decide, we need a way to expose it via an iterator. We are currently looping through tokens using your awesome `split_front` and `split_back` methods, which is a hell of a lot better than it was before (with all the allocations and such) but it feels like we could have a more purpose-built solution to needing to walk a pointer forward/backward.

> Yet another option, if the offset is intended so we can find a token in the Pointer, is to use a slice instead.

Sorry, I don't understand the option on this one.

Reply:
https://github.com/chanced/jsonptr/issues/42

Reply:
Yeah that makes sense. My main worry with using token indices is the need to traverse the string to count them — which is perfectly reasonable to have in the API, but if we use it internally the cost compounds. I like the idea of doing this via an iterator. Probably makes it easier to change the implementation later if necessary.